helper/server/downstream_test.go

```diff
@@ -49,10 +52,15 @@ func TestDownstreamServer(t *testing.T) {
 			Polling:      true,
 		})
 		if err != nil {
-			t.Fatal(err)
+			errs <- err
 		}
 	}()
 
+	err = <-errs
```

Comment:
I don't think this resembles the logic from before as the function does not immediately return